From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Matthew Feickert <matthew.feickert@cern.ch>
Date: Sat, 3 Oct 2026 15:27:35 -0600
Subject: [PATCH] Fix herwig-config exit status and use grep -E

herwig-config ends with the check for its last option

    test -n "$tmp" && echo @THEPEGLIB@

so its exit status is 1 for every option but --ldlibs, including --help.
Exit with 0 explicitly so that the exit status can be relied on.

egrep is obsolescent and GNU grep >= 3.8 warns on every use

    egrep: warning: egrep is obsolescent; using grep -E

which herwig-config does once per option that it knows. Use grep -E
instead.
---
 src/herwig-config.in | 17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

diff --git a/src/herwig-config.in b/src/herwig-config.in
index 31ef2e5..d2e10ba 100644
--- a/src/herwig-config.in
+++ b/src/herwig-config.in
@@ -5,7 +5,7 @@ prefix=@prefix@
 exec_prefix=@exec_prefix@
 datarootdir=@datarootdir@
 
-tmp=$(echo $* | egrep -- '--\<help\>|-\<h\>')
+tmp=$(echo $* | grep -E -- '--\<help\>|-\<h\>')
 if test $# -eq 0 || test -n "$tmp"; then
     echo "herwig-config: configuration tool for the Herwig event generator"
     echo "               http://projects.hepforge.org/herwig/"
@@ -24,24 +24,25 @@ if test $# -eq 0 || test -n "$tmp"; then
     echo
 fi
 
-tmp=$(echo "$*" | egrep -- '--\<prefix\>')
+tmp=$(echo "$*" | grep -E -- '--\<prefix\>')
 test -n "$tmp" && echo $prefix
 
-tmp=$(echo "$*" | egrep -- '--\<datadir\>')
+tmp=$(echo "$*" | grep -E -- '--\<datadir\>')
 test -n "$tmp" && echo @datadir@/@PACKAGE_TARNAME@
 
-tmp=$(echo "$*" | egrep -- '--\<libdir\>')
+tmp=$(echo "$*" | grep -E -- '--\<libdir\>')
 test -n "$tmp" && echo @libdir@/@PACKAGE_TARNAME@
 
-tmp=$(echo "$*" | egrep -- '--\<includedir\>')
+tmp=$(echo "$*" | grep -E -- '--\<includedir\>')
 test -n "$tmp" && echo @includedir@
 
-tmp=$(echo "$*" | egrep -- '--\<cppflags\>')
+tmp=$(echo "$*" | grep -E -- '--\<cppflags\>')
 test -n "$tmp" && echo -I@includedir@ @THEPEGINCLUDE@ @BOOST_CPPFLAGS@ @GSLINCLUDE@
 
-tmp=$(echo "$*" | egrep -- '--\<ldflags\>')
+tmp=$(echo "$*" | grep -E -- '--\<ldflags\>')
 test -n "$tmp" && echo @THEPEGLDFLAGS@
 
-tmp=$(echo "$*" | egrep -- '--\<ldlibs\>')
+tmp=$(echo "$*" | grep -E -- '--\<ldlibs\>')
 test -n "$tmp" && echo @THEPEGLIB@
 
+exit 0
